src/util/animate_color.ts

```diff
@@ -46,40 +62,46 @@ const defaultColorEasing = (currentTime, duration) =>
  * @returns {Function} abort function
  */
 export function animateColor(
-  fromColor,
-  toColor,
+  fromColor: string,
+  toColor: string,
   duration = 500,
   {
     colorEasing = defaultColorEasing,
     onComplete,
     onChange,
-    ...restOfOptions
+    abort,
+    target,
+  }: {
+    colorEasing?: ColorEasing;
+    onComplete?: OnColorChangeCallback;
+    onChange?: OnColorChangeCallback;
+    abort?: AbortCallback;
+    target?: unknown;
   } = {}
-) {
+): CancelFunction {
   const startColor = new Color(fromColor).getSource(),
     endColor = new Color(toColor).getSource();
   return animate({
-    ...restOfOptions,
+    target,
```

Comment:
can you explain this change?
target seem to have been introduced in this PR, and removing ...restOfOptions cuts away using animate_color as animate ( delay and onStart are missing to make an example ) requiring changes in 2 places when something else gets added to animate

Reply:
target was not introduced here, it was just apparently never documented. I found that out when I ran the tests. Look at the animation.minix.ts _animate function and the animation unit tests, as well as the animation_registry.ts.

Reply:
d2075e6

Reply:
I get target now, but the rest of the question still stands.
Why did you choose to remove ...restOfOptions ? animate color takes every option than animate does, and so we should pass down with a transparent pass through

Reply:
I chose to remove it because I didn't know it was intended for the object to just be all other animation options. I can take a look and refactor it.  I'm thinking that for this PR I'm going to take animate_color out. I'll submit another PR since the changes are too big.
Like I said AFAIK there's no easy way to generify `animate` so that it works well with the current animate color.

Reply:
there is no easy way no, and that is why it was a different function.
Normal easing functions do not really make sense for colors.
The issue is that i still convinced that since we introduced array support for animate we did break animate_color